controller/src/beerocks/master/son_management.cpp

```diff
@@ -1505,8 +1505,49 @@ void son_management::handle_bml_message(Socket *sd,
     }
     case beerocks_message::ACTION_BML_WIFI_CREDENTIALS_SET_REQUEST: {
         LOG(TRACE) << "ACTION_BML_WIFI_CREDENTIALS_SET_REQUEST";
-        // TODO: trigger auto-config, probaly it will be a good idea to change the message to
-        // "ACTION_BML_UPDATE_CONFIGURATION_REQUEST"
+        son::wireless_utils::sBssInfoConf wifi_credentials;
+        uint32_t ret = 1;
+        auto request =
+            beerocks_header->addClass<beerocks_message::cACTION_BML_WIFI_CREDENTIALS_SET_REQUEST>();
+        if (!request) {
+            LOG(ERROR) << "addClass cACTION_BML_WIFI_CREDENTIALS_SET_REQUEST failed";
+            return;
+        }
+
+        wifi_credentials.ssid                = request->ssid();
+        wifi_credentials.network_key         = request->network_key();
+        wifi_credentials.authentication_type = WSC::eWscAuth(request->authentication_type());
+        wifi_credentials.encryption_type     = WSC::eWscEncr(request->encryption_type());
+        wifi_credentials.bss_type = WSC::eWscVendorExtSubelementBssType(request->bss_type());
+
+        auto op_ch = request->operating_channels();
+        for (int i = 0; i < request->operating_channel_size(); i++) {
+            wifi_credentials.operating_class.push_back(uint16_t(op_ch[i]));
+        }
+
+        auto al_mac_addr = network_utils::mac_to_string(request->al_mac());
+        auto ires        = database.get_all_connected_ires();
+
+        if (ires.find(al_mac_addr) != ires.end()) {
```

Comment:
In this case, the condition shouldn't be there. It should be possible to configure credentials for an agent before it starts up. Note that add_bss_info_configuration does not have such a check and doesn't add anything to the node - the bssinfo database is separate.

Actually, it was probably wrong to put the credentials in the database to begin with. It's configuration information, not topology information. But that's not for this PR.